tensorboard/plugins/pr_curve/pr_curves_plugin.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+import numpy as np
+import six
+from werkzeug import wrappers
+
+from tensorboard.backend import http_util
+from tensorboard.plugins import base_plugin
+from tensorboard.plugins.pr_curve import metadata
+
+
+class PrCurvesPlugin(base_plugin.TBPlugin):
+  """A plugin that serves PR curves for individual classes."""
+
+  plugin_name = metadata.PLUGIN_NAME
+
+  def __init__(self, context):
+    """Instantiates a PrCurvesPlugin.
+    Args:
+      context: A base_plugin.TBContext instance. A magic container that
+        TensorBoard uses to make objects available to the plugin.
+    """
+    self._multiplexer = context.multiplexer
+
+  @wrappers.Request.application
+  def pr_curves_route(self, request):
+    """A route that returns a JSON mapping between runs and PR curve data.
+
+    Returns:
+      Given a tag and a comma-separated list of runs (both stored within GET
+      parameters), fetches a JSON object that maps between run name and objects
+      containing data required for PR curves for that run. Runs that either
+      cannot be found or that lack tags will be excluded from the response.
+    """
+    runs = request.args.getlist('run')
+    tag = request.args.get('tag')
+
+    response_mapping = {}
+    for run in runs:
+      try:
+        tensor_events = self._multiplexer.Tensors(run, tag)
+      except KeyError:
+        return http_util.Respond(
+            request,
+            'No PR curves could be fetched for run %r and tag %r' % (run, tag),
+            'text/plain',
+            400)
+      
+      response_mapping[run] = [
+          self._process_tensor_event(e) for e in tensor_events]
+
+    # We convert the tensor data into a JSON-able response.
+    return http_util.Respond(request, response_mapping, 'application/json')
+
+  @wrappers.Request.application
+  def tags_route(self, request):
```

Comment:
You'll need to update this to return a run-to-tag-to-tag-info map, as in the other plugins. Then, you can pass this to `tf-card-heading` on the frontend. (The frontend part can be a TODO for now because of resolution, but we should get the backend right on the first go. Take a look at the `runToTagInfo` adaptations in the scalar dashboard, for instance.)